templates/ra_guac_autoscale_public_alb.template.cfn.yaml

```diff
@@ -0,0 +1,623 @@
+AWSTemplateFormatVersion: 2010-09-09
+Conditions:
+  UseACM: !Equals
+    - !Ref SslCertificateService
+    - ACM
+  UseAmiLookup: !Equals
+    - !Ref AmiId
+    - ''
+  UseAutoUpdateStack: !Not
+    - !Equals
+      - !Ref UpdateSchedule
+      - ''
+  UseBrandText: !Not
+    - !Equals
+      - !Ref BrandText
+      - ''
+  UseScalingPolicy: !Or
+    - !Equals
+      - !Ref ScaleUpSchedule
+      - ''
+    - !Equals
+      - !Ref ScaleDownSchedule
+      - ''
+  UseScheduledAction: !And
+    - !Not
+      - !Equals
+        - !Ref ScaleUpSchedule
+        - ''
+    - !Not
+      - !Equals
+        - !Ref ScaleDownSchedule
+        - ''
+  UseURL1: !Not
+    - !Equals
+      - !Ref URL1
+      - ''
+  UseURL2: !Not
+    - !Equals
+      - !Ref URL2
+      - ''
+  UseURLText1: !Not
+    - !Equals
+      - !Ref URLText1
+      - ''
+  UseURLText2: !Not
+    - !Equals
+      - !Ref URLText2
+      - ''
+Description: This templates deploys Guacamole (Guac) instances in an AutoScale Group behind an ALB
+Mappings:
+  InstanceTypeMap:
+    'Fn::Transform':
+      Name: 'AWS::Include'
+      Parameters:
+        Location: 's3://app-chemistry/snippets/instance_type_map.snippet.cfn.yaml'
+Metadata:
+  Version: 0.3.1
+  cfn-lint:
+    config:
+      ignore_checks:
+        - E7001
+        - W1011
+Outputs:
+  AlbSecurityGroupId:
+    Description: Security Group ID for the Guacamole Application Load Balancer
+    Value: !Ref AlbSecurityGroup
+  Ec2SecurityGroupId:
+    Description: Security Group ID for Guacamole instances
+    Value: !Ref Ec2SecurityGroup
+  LoadBalancerDns:
+    Description: DNS name for the ALB
+    Value: !Sub ${ALB.DNSName}
+  LoadBalancerName:
+    Description: Name of the ALB
+    Value: !Ref ALB
+  TargetGroupName:
+    Description: Name of the ALB Target Group
+    Value: !Ref ALBTargetGroup
+Parameters:
+  AmiId:
+    Default: ''
+    Description: (Optional) AMI ID -- will supersede Lambda-based AMI lookup using AmiNameSearchString
+    Type: String
+  AmiNameSearchString:
+    Default: amzn-ami-hvm-2017.09*.2017*-x86_64-gp2
```

Comment:
I get that this is just a default, but we should probably update it to a current year

Reply:
updating to "amzn-ami-hvm-2018.03*.2018*-x86_64-gp2". Stack created successfully with the new image search string